src/openforms/formio/dynamic_config/tests/test_dynamic_options.py

```diff
@@ -743,3 +753,42 @@ def test_different_label_key_options(self):
                 {"label": "Test2", "value": "987654321"},
             ],
         )
+
+
+class TestDynamicConfigAddingOptionsForRequest(SubmissionsMixin, APITestCase):
+    @tag("gh-2895")
+    def test_overwrite_html_in_content_component(self):
+        """Assert that style tag remains and nonce is added to content component"""
+        configuration = {
+            "components": [
+                {
+                    "key": "content",
+                    "type": "content",
+                    "html": '<p><span style="color:#e64c4c;">Test</span></p>',
+                },
+            ]
+        }
+        form = FormFactory.create()
+        form_definition = FormDefinitionFactory.create(
+            configuration=configuration,
+        )
+        step = FormStepFactory.create(form=form, form_definition=form_definition)
+        submission = SubmissionFactory.create(form=form)
+
+        self._add_submission_to_session(submission)
+        endpoint = reverse(
+            "api:submission-steps-detail",
+            kwargs={
+                "submission_uuid": submission.uuid,
+                "step_uuid": step.uuid,
+            },
+        )
+        user = SuperUserFactory.create()
+        self.client.force_authenticate(user=user)
+
+        response = self.client.get(endpoint, HTTP_X_CSP_NONCE="dGvsa==")
+        formio_component = response.json()["formStep"]["configuration"]["components"][0]
+        html = formio_component["html"]
+
+        self.assertIn("nonce", html)
+        self.assertIn("color:#e64c4c", html)
```

Comment:
I think it might be good to make the test and the fix a bit more specific.

For example, if you change the component in this test to be `<p><span style="color:#e64c4c;">Test nonce</span></p>` you see that at the end the inline style is not properly changed into a `<style>` tag and the test still succeeds.

So maybe in the fix check that the nonce is in the inline style and in the test check that the nonce is in the id and there is a `style` tag present?